### PR TITLE
Add lambda initializers

### PR DESCRIPTION
**Pull request by cmpute**

### CODE REVIEW

**Review by b1f6c1c4 — CHANGES_REQUESTED** [review]
Unfortunately, I consider your `function-empty.hpp` and `lambda-empty.hpp` identical to the existing `direct-empty.hpp` because the function call of `identity` is doing EXTRA work -- in other words, changing `identity(T())` to `T()` will same the exact same effect AND the same mechanism.

Note that `function-implicit.hpp` is not identical to `direct-implicit.hpp` because they are two different mechanisms -- implicit conversion at return statement, and implicit conversion at function call.

As for lambda function calls, I consider they DIFFERENT from regular function calls because it's actually some `operator()` call of an anonymous functor type. It's member function call.

To summarize, please remove the two snippets using `T()`.
